visualization/src/pcl_visualizer.cpp

```diff
@@ -2980,10 +2980,10 @@ pcl::visualization::PCLVisualizer::updateColorHandlerIndex (const std::string &i
     return (false);
   }
 
-  int color_handler_size = int (am_it->second.color_handlers.size ());
-  if (index >= color_handler_size)
+  size_t color_handler_size = am_it->second.color_handlers.size ();
+  if (!(size_t(index) < color_handler_size))
   {
-    pcl::console::print_warn (stderr, "[updateColorHandlerIndex] Invalid index <%d> given! Maximum range is: 0-%lu.\n", index, static_cast<unsigned long> (am_it->second.color_handlers.size ()));
+    pcl::console::print_warn (stderr, "[updateColorHandlerIndex] Invalid index <%d> given! Index must be less than %d.\n", index, int(color_handler_size));
```

Comment:
Actually, why casting here? Just use `%zu` format.

Reply:
`%zu` is officially available only from C++11 forwards. is it okay to use?

Reply:
AFAIK it's ANSI C99 feature. But your question reminded me of the issues we had with MSVC (#588). So better use `%lu`.

Reply:
Yes, exactly it is ANSI C99 feature. But ISO C++03 has not been rebased on C99, so also `%zu` is missing. This has been done in C++11 (rebase on C99) and there is `%zu` along with other. Only C++14 is finally based on C11.

However in real world `%zu` is pretty much fine on all linuxes, and some recent VS. macs should be fine too. If you turn on `-pedantic` it should be reported as standard violation.

Reply:
Oh, good to know!